src/db/SQL.js

```diff
@@ -322,7 +322,7 @@ define([
 						}
 						return ['(', parts.join(sqlOperators[filter.type]), ')'];
 					case 'in':
-						if (value.length === 0) {
+						if (!(typeof value === 'object') && value.length === 0) {
```

Comment:
It was fixing the issue, that since the `newCollection` object created in `_createSubCollection` was being created from the constructor of the class, it had a length property with value 0, since I fixed `_createSubCollection` to use `Object.create` on the prototype per your recommendation I was able to undo this change.